src/pages/DoctorDashboard.tsx

```diff
@@ -9,12 +10,156 @@ import {
     Navigate,
     Outlet,*/
 } from "react-router-dom";
+import {Vaccination, Visit, AuthContextType} from "../components/types";
+import {addDays, getBeginningOfWeek} from "../components/dateUtils";
+import DatePicker from 'react-date-picker';
+import { getSlots } from '../logic/doctorAPI';
+import { useAuth } from '../App';
+
 function DoctorDashboard() {
     const navigate = useNavigate()
+    let today = getBeginningOfWeek(new Date());
+    today.setHours(0,0,0,0);
+    const [startDate, onStartDateChange] = useState<Date>(today);
+    const [endDate, onEndDateChange] = useState<Date>(addDays(today, 6));
+    const [reserved, setReserved] = useState<string>('-1');
+    const [page, setPage] = useState<number>(1);
+    const [maxPage, setMaxPage] = useState<number>(10);
+    const radios = [
+        { name: 'All', value: '-1' },
+        { name: 'Free', value: '0' },
+        { name: 'Reserved', value: '1' },
+    ];
+    let auth: AuthContextType = useAuth();
+    const sampleInfo: Visit[] = [
+        {
+            date: new Date("2022-03-29T08:00:00Z"),
+            id: 0,
+            vaccination: null,
+        },
+        {
+            date: new Date("2022-03-30T08:00:00Z"),
+            id: 1,
+            vaccination: {
+                id: 0,
+                status: "Planned",
+                patient: {
+                    id: 0,
+                    firstName: "Adam",
+                    lastName: "Abacki",
+                    pesel: "12345678901",
+                    email: "email@example.com",
+                    address: {
+                        id: 0,
+                        city: "Warszawa",
+                        zipCode: "01-234",
+                        street: "Przykładowa",
+                        houseNumber: 1,
+                        localNumber: null,
+                    }
+                },
+                vaccine: {
+                    id: 0,
+                    name: "Phiser",
+                    disease: "Covid-19",
+                    requiredDoses: 0,
+                },
+            }
+        },
+        {
+            date: new Date("2022-03-29T10:00:00Z"),
+            id: 0,
+            vaccination: null,
+        },
+    ];
+    function getVisits(start: Date | null, finish: Date | null, onlyReserved: string, page: number): Visit[] {
+        getSlots(start, finish, onlyReserved, auth.token, page).then((response) => {
+            if(response.ok)
```

Comment:
Tutaj już nie musisz sprawdzać czy response.ok, bo sprawdziłeś to już w funkcji getSlots. I wejdzie ci do tej funkcji w argumencie then tylko jeśli będzie "respose.ok".